client/payment-details/payment-method/giropay/index.js

```diff
@@ -0,0 +1,118 @@
+/** @format **/
+
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies.
+ */
+import PaymentDetailsPaymentMethodDetail from '../detail';
+
+/**
+ * Extracts and formats payment method details from a charge.
+ *
+ * @param {Object} charge The charge object.
+ * @return {Object}       A flat hash of all necessary values.
+ */
+const formatPaymentMethodDetails = ( charge ) => {
+	const { billing_details: billingDetails, payment_method: id } = charge;
+
+	// eslint-disable-next-line camelcase
+	const { bank_name, bic } = charge.payment_method_details.giropay;
+
+	const { name, email, formatted_address: formattedAddress } = billingDetails;
+
+	return {
+		id,
+		// eslint-disable-next-line camelcase
+		bankName: bank_name,
+		bic,
+		name,
+		email,
+		formattedAddress,
+	};
+};
+
+/**
+ * Placeholders to display while loading.
+ */
+const paymentMethodPlaceholders = {
+	bankName: 'bank name placeholder',
+	bic: 'bic placeholder',
+	id: 'id placeholder',
+	name: 'name placeholder',
+	email: 'email placeholder',
+	formattedAddress: 'address placeholder',
+	country: 'country placeholder',
+};
+
+const GiropayDetails = ( { charge = {}, isLoading } ) => {
```

Comment:
(minor) since there's already a check on `charge` on L53 there should be no need for `= {}`.
Or can `charge` be falsy otherwise? In that case I think we can change L53 from `charge && charge.payment_method_details` to `charge.payment_method_details`, since there's `= {}`

Reply:
I went with the second option to only check against `charge.payment_method_details`, thanks!